typescript-decorators/src/decorators.ts

```diff
@@ -0,0 +1,31 @@
+import { FastifyReply, FastifyRequest } from 'fastify'
+import { IncomingMessage, ServerResponse } from 'http'
+import { instance } from './instance'
```

Comment:
In typically fastify apps, we do not recommend exporting the fastify instance as a module global.
A fastify application can only be booted once, and in unit test we really need to be able to boot an application multiple times.

Could this be refactored to register the classes instead?

Instead of calling `instance.get()` within the decorator, I think we should provide some way to convert these with a:

```js
instance.register(magicalRoutes)
```

Reply:
Sorry, didn't get it..

Reply:
The problem with exposing the fastify instance as a global in the `require.cache` is that you cannot instantiate it more than once. You are going to reuse the same instance across multiple tests, and this is going to create flakyness in your application. It is not recommended.

In order for this pattern to work well, you need something like https://github.com/fastify/fastify-autoload that loads the files and reads the definitions, and then add all the routes. This needs to happen _every time_ the plugin is registered.

Given how fastify works, the following might be more appropriate:

```ts
@GET({
    url: '/get'
}
   async function handle(): Promise<any> {
        return {
            message: 'GET request work!'
        }
    }
```

Reply:
I'll take a look later today